packages/web-client/app/components/card-pay/balance-view-banner/index.ts

```diff
@@ -0,0 +1,66 @@
+import Component from '@glimmer/component';
+import BN from 'bn.js';
+
+import { MerchantInfo } from '@cardstack/web-client/resources/merchant-info';
+import { MerchantSafe, Safe } from '@cardstack/cardpay-sdk';
+import { useResource } from 'ember-resources';
+import {
+  BridgedTokenSymbol,
+  TokenDisplayInfo,
+  getUnbridgedSymbol,
+} from '@cardstack/web-client/utils/token';
+
+interface BalanceViewBannerComponentArgs {
+  walletAddress: string;
+  safe: Safe;
+  token: BridgedTokenSymbol;
+}
+
+class BalanceViewBannerComponent extends Component<BalanceViewBannerComponentArgs> {
+  supportedSafeTypes = ['depot', 'merchant'];
+
+  get tokenBalance(): BN {
+    let safe = this.args.safe;
+    let tokenSymbol = this.args.token;
+    let unbridgedSymbol = getUnbridgedSymbol(tokenSymbol);
+    // SDK returns bridged token symbols without the CPXD suffix
+
+    let balance = safe.tokens.find(
+      (token) => token.token.symbol === unbridgedSymbol
+    )?.balance;
+
+    return balance ? new BN(balance) : new BN(0);
+  }
+
+  get safeIcon() {
+    if (this.supportedSafeTypes.includes(this.args.safe.type)) {
+      return this.args.safe.type;
+    } else {
+      return null;
+    }
+  }
+
+  get tokenIcon() {
+    return TokenDisplayInfo.iconFor(this.args.token);
+  }
+
+  get summaryBalanceLabel() {
+    if (this.args.safe.type === 'merchant') {
+      if (this.merchantInfo.loading) {
+        return `…`;
```

Comment:
Would it be good to show "Merchant" while the merchant info is loading? Also why is there a difference in capitalization between merchant and depot 🤔